### PR TITLE
Do not link bin scripts from dev dependencies with `--production`

### PR DESCRIPTION
**Pull request by bruderstein**
`bin` scripts for `devDependencies` when installing with `--production` should not be linked

**Summary**

When installing with `--production`, entries in `devDependencies` are not installed, but their `bin` scripts were linked to the `node_modules/.bin` directory. The symlinks were actually generated to the cache.

This change means that `bin` scripts are no longer added for devDependencies when in `--production` (actually for `ignore`d dependencies)

**Test plan**

In an empty directory
```
yarn init -y
yarn add eslint --dev
yarn add gulp
rm -rf node_modules
yarn install --production
ls -l node_modules/.bin
```

This shows the `eslint` binary linked to the cache. On my mac, that looks as follows:
```
$ ls -l node_modules/.bin
total 16
lrwxr-xr-x  1 dxxxxxxxxx  staff  107 Jan 29 21:24 eslint -> ../../../../../Library/Caches/Yarn/npm-eslint-3.14.1-8a62175f2255109494747a1b25128d97b8eb3d97/bin/eslint.js
lrwxr-xr-x  1 dxxxxxxxxx  staff   19 Jan 29 21:24 gulp -> ../gulp/bin/gulp.js
```

With the PR applied, only the `gulp` binary is linked in the `.bin` directory.

Added a test to test exactly this scenario.

This actually manifested itself as a problem in our CI environment, where the cache is owned by a different owner, so the `chmod` failed.

### CODE REVIEW

**Comment by bruderstein** [comment]
Failing test runs seem to be timing?

**Comment by bruderstein** [comment]
Fixes #2214

File: __tests__/fixtures/install/install-production-bin/package.json [hunk 1]
```diff
@@ -0,0 +1,9 @@
+{
+  "devDependencies": {
```
Comment (bestander) [inline __tests__/fixtures/install/install-production-bin/package.json]: This is an excessively large node_modules that gets installed when the test is running.
Could you find lighter ones for the test?

**Comment by bruderstein** [comment]
@bestander fixed that - thanks for the tip! (Switched to `rimraf` and `touch`) The added test now takes 2ms \o/

The failure on travis is actually the other `--production` test timing out. Happy to switch those dependencies too, but that maybe shouldn't be part of this PR?

**Comment by bestander** [comment]
Thanks, @bruderstein.
Would be great if you send a PR to fix the other test!

**Pull request merged**